packages/material-ui-styles/src/withStyles/withStyles.d.ts

```diff
@@ -43,9 +50,12 @@ export interface CreateCSSProperties<Props extends object = {}>
  */
 export type StyleRules<Props extends object = {}, ClassKey extends string = string> = Record<
   ClassKey,
-  IsEmptyInterface<Props> extends true
-    ? CSSProperties | (() => CSSProperties)
```

Comment:
Not sure what the thunk without arguments was used for. I originally thought this was for theme based styles but then we would still need an argument. And we already take care of the theme in  `StyleRulesCallback`

Reply:
It was added here, https://github.com/mui-org/material-ui/pull/16200
tl;dr: backwards compatibility

Reply:
Seems like it wasn't necessary to begin with since the added tests are passing in 3.2.4 as well as 3.7.4.

Reply:
It was meant to handle this demo https://github.com/mui-org/material-ui/blob/78f5ec8fff38a6a72fb4f56049b891910f808c76/docs/src/pages/customization/components/DynamicInlineStyle.tsx#L6
Which you've changed, so everything passes

Reply:
Thanks for clarifying.